packages/gatsby-theme-docs/src/hooks/use-top-menu-items.js

```diff
@@ -0,0 +1,32 @@
+import { useStaticQuery, graphql } from 'gatsby';
+
+const useTopMenuItems = () => {
+  const data = useStaticQuery(graphql`
+    query GetTopMenuItems {
+      allTopMenuYaml {
+        nodes {
+          menuTitle
+          items {
+            href
+          }
+        }
+      }
+    }
+  `);
+
+  const siteContextMap = new Map();
+  data.allTopMenuYaml.nodes.forEach((node) => {
+    node.items.forEach((item) => {
+      if (item.href.startsWith('/../')) {
+        const minisiteSegment = item.href.split('/')[2];
+        if (minisiteSegment) {
+          siteContextMap.set(minisiteSegment, node.menuTitle);
+        }
+      }
```

Comment:
Can you explain what this logic is for? 🤔

Reply:
sure
```
      if (item.href.startsWith('/../')) { // this excludes absolute urls 
        const minisiteSegment = item.href.split('/')[2]; // this gets the first non ".." segment of the url, normally matching the minisite baseUrl segment
        if (minisiteSegment) {
          // creates a map mapping a specific minisite to his ancestor as defined in the `top-menu.yaml`
          siteContextMap.set(minisiteSegment, node.menuTitle);
        }
      }
```

The idea behind this is that depending on the minisite I'm on, I've got a reverse map `siteContextMap` where I can lookup if the minisite belongs to Composable Commerce or Composable Frontend (or in main general look up what's the ancestor of a given minisite).